### PR TITLE
Adding a new point related to the extension's performance

### PR DESCRIPTION
**Pull request by eduard13**
## Purpose of this pull request

This pull request (PR) adds a new important point that is related to the extension's performance.

## Affected DevDocs pages

- https://devdocs.magento.com/guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.html
- https://devdocs.magento.com/guides/v2.2/ext-best-practices/extension-coding/security-performance-data-bp.html
- https://devdocs.magento.com/guides/v2.3/ext-best-practices/extension-coding/security-performance-data-bp.html

### CODE REVIEW

**Comment by devops-devdocs** [comment]
An admin must run tests on this PR before it can be merged.

File: guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.md [hunk 1]
```diff
@@ -51,3 +51,28 @@ For a list of discouraged low-level functions, review the [`Magento2/Sniffs/Func
 
 ### Always encrypt sensitive data or configurations
   Never store sensitive information in clear text within a resource that might be accessible to another control sphere. This type of information should be encrypted or otherwise protected.
+
+### Avoid running unneeded logic
```
Comment (rogyar) [inline guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.md]: Please, adjust the topic title slightly
```suggestion
### Avoid unnecessary logic execution
```

File: guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.md [hunk 2]
```diff
@@ -51,3 +51,28 @@ For a list of discouraged low-level functions, review the [`Magento2/Sniffs/Func
...
+
+  Make sure that you never run some code that it won't be used in the next step. <br><br>
```
Comment (rogyar) [inline guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.md]: Do we need `<br><br>` here? We try to use clean Markdown wherever it's possible.

Comment (eduard13) [reply guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.md]: Nope, we don't. Removed them.

File: guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.md (hunk 1 above)
Comment (eduard13) [reply guides/v2.1/ext-best-practices/extension-coding/security-performance-data-bp.md]: Adjusted

**Comment by eduard13** [comment]
Hi @rogyar, the suggestions were successfully applied. Could you please check the latest changes?
Thank you.

**Comment by rogyar** [comment]
Hi @eduard13. Awesome, thank you

**Comment by dobooth** [comment]
running tests

**Comment by dobooth** [comment]
running tests

**Pull request merged**